connector.go

```diff
@@ -312,10 +312,43 @@ func (c *Connector) RegisterProtos(msgType kafkautils.MsgType, protos ...proto.M
 
 	tt := c.buildTopicTypes(msgType, protos...)
 
-	err := protoregistry.RegisterDynamicTopics(c.protoRegistryHost, tt, msgType)
-	if err != nil {
-		log.Fatal().Err(err).Msg("failed to register dynamic topics")
-	}
+	// retry it with exponential decay
+	go func() {
```

Comment:
may be we can put this whole function as `RetryWithExponentialDecay()` and pass the `RegisterDynamicTopics()` to it? otherwise this will prompt people to copy paste this large chunk of code

Reply:
Why would people need to copy paste this code?

Reply:
this is example code. afaik people are supposed to copy paste / fork this and create their own connectors.

Reply:
nvm I thought it was in example code 🤦